lib/perl/Genome/InstrumentData/AlignmentResult/Merged.pm

```diff
@@ -219,25 +226,65 @@ sub create {
         for my $alignment (@alignments) {
             $alignment->add_user(user => $self, label => 'uses');
         }
-
-        return 1;
-    };
-    if (my $error = $@) {
-        $tx->rollback();
-        die $error;
-    } elsif ($rv ne 1) {
+    }
+    catch {
         $tx->rollback();
-        $self->error_message('Unexpected return value: ' . $rv);
-        die $self->error_message;
-    } else {
-        $tx->commit();
+        die $self->error_message('Merge failed due to error: ' . $_);
+    }
+    #The transaction commit and temporary allocation deletion need to
+    #be in different finally blocks.
+    finally {
+        unless ($_) {
```

Comment:
Thanks very much for all the insight here. We've pushed a commit that moves the commit up to the try block. We agree this seems much cleaner!